File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
object ScrabbleScore {

    fun scoreLetter(c: Char): Int {
        when {
            "AEIOULNRST".contains(c) -> return 1
            "DG".contains(c) -> return 2
            "BCMP".contains(c) -> return 3
            "FHVWY".contains(c) -> return 4
            "K".contains(c) -> return 5
            "JX".contains(c) -> return 8
            else -> return 10
        }
    }

    fun scoreWord(word: String): Int {
        if (word.length == 0) {
            return 0
        }
        return word.toUpperCase().map { c -> scoreLetter(c) }.reduce { x, y -> x + y }
    }
}

File: go/scrabble-score/scrabble_score.go
// Package scrabble provides a function for calculating scrabble scores
package scrabble

import (
	s "strings"
)

var scoreLetter = map[string]int{"AEIOULNRST": 1, "DG": 2, "BCMP": 3, "FHVWY": 4, "K": 5, "JX": 8, "QZ": 10}

// Score returns the scrabble score for the given input string
func Score(word string) int {
	total := 0
	for i := 0; i < len(word); i++ {
		upperLetter := s.ToUpper(string(word[i]))
		for group, score := range scoreLetter {
			if s.Contains(group, upperLetter) {
				total += score
				break
			}
		}
	}
	return total
}
